schema/list_test.go

```diff
@@ -182,20 +182,27 @@ func TestListVerifyCompatibility(t *testing.T) {
 	assert.Error(t, intListSchema.ValidateCompatibility([]any{schema.NewStringSchema(nil, nil, nil)}))
 }
 
-func TestUnserialize_Idempotent(t *testing.T) {
+func TestList_UnSerialize_Reversible(t *testing.T) {
 	listType := schema.NewTypedListSchema[string](
 		schema.NewStringSchema(
 			nil,
 			nil,
 			nil,
 		),
-		schema.IntPointer(2),
+		schema.IntPointer(3),
 		nil,
 	)
-	serializableInput := []any{"foo", "bar"}
-	unserialized, err := listType.Unserialize([]any{"foo", "bar"})
+	serializableInput := []any{"foo", "bar", "baz"}
+	unserialized, err := listType.Unserialize(serializableInput)
 	assert.NoError(t, err)
 	serialized, err := listType.Serialize(unserialized)
 	assert.NoError(t, err)
-	assert.Equals(t, serialized.([]any), serializableInput)
+	unserialized2, err := listType.Unserialize(serialized)
+	assert.NoError(t, err)
+	serialized2, err := listType.Serialize(unserialized2)
+	assert.NoError(t, err)
+
+	// test unserialize and serialize are reversible
+	assert.Equals(t, unserialized2, unserialized)
+	assert.Equals(t, serialized2, serialized)
```

Comment:
Similar to the previous two comments, I don't think you need to reserialize the result of the second unserialization.

However, I think you _should_ compare the result of the first reserialization to the the original, which you change here has removed!  (Instead, you're comparing the results of the two reserializations, which doesn't cover that ground....)

The same comment applies to `TestMap_UnSerialize_Reversible()` in `schema/map_test.go`, and to the tests in `schema/object_test.go`, and probably the subsequent ones.

Reply:
I think that kind of test would be better in the underlying type rather than the container type. Also, like y'all said, there are multiple, unique inputs that can lead to `serialized`, so I don't think asserting equality on the `input` and `serialized` is a helpful test.